src/diff/index.js

```diff
@@ -252,10 +241,9 @@ function diffElementNodes(dom, newVNode, oldVNode, context, isSvg, excessDomChil
 		// we created a new parent, so none of the previously attached children can be reused:
 		excessDomChildren = null;
 	}
-	newVNode._dom = dom;
 
 	if (newVNode.type===null) {
-		if ((d===null || dom===d) && newVNode.text!==oldVNode.text) {
+		if ((originalDom==null || dom===originalDom) && newVNode.text!==oldVNode.text) {
```

Comment:
since `dom` can never be `null` here, checking `dom===originalDom` implies `originalDom` is also non-null, so we can drop that condition entirely:

```suggestion
		if (dom===originalDom && newVNode.text!==oldVNode.text) {
```